source/auth/tests/connection-string.json

```diff
@@ -1,279 +1,344 @@
+        "password": null,
+        "db": "$external"
+      },
+      "options": {
+        "authmechanism": "GSSAPI"
+      }
+    },
+    {
+      "description": "should recognise the mechanism where authSource $external (GSSAPI)",
+      "uri": "mongodb://user%40DOMAIN.COM@localhost/foo?authMechanism=GSSAPI&authSource=$external",
+      "hosts": null,
+      "valid": true,
+      "warning": false,
+      "auth": {
+        "username": "user@DOMAIN.COM",
+        "password": null,
+        "db": "$external"
+      },
+      "options": {
+        "authmechanism": "GSSAPI"
+      }
+    },
+    {
+      "description": "should accept generic mechanism property (GSSAPI)",
+      "uri": "mongodb://user%40DOMAIN.COM@localhost/?authMechanism=GSSAPI&authMechanismProperties=SERVICE_NAME:other,CANONICALIZE_HOST_NAME:true",
+      "hosts": null,
+      "valid": true,
+      "warning": false,
+      "auth": {
+        "username": "user@DOMAIN.COM",
+        "password": null,
+        "db": "$external"
+      },
+      "options": {
+        "authmechanism": "GSSAPI",
+        "authmechanismproperties": {
+          "SERVICE_NAME": "other",
+          "CANONICALIZE_HOST_NAME": true
+        }
+      }
+    },
+    {
+      "description": "should recognise password if SSPI is supported (GSSAPI)",
+      "uri": "mongodb://user%40DOMAIN.COM:password@localhost/foo?authMechanism=GSSAPI&authSource=$external",
+      "hosts": null,
+      "valid": true,
+      "warning": false,
+      "auth": {
+        "username": "user@DOMAIN.COM",
+        "password": "password",
+        "db": "$external"
+      },
+      "options": {
+        "authmechanism": "GSSAPI"
+      }
+    },
+    {
+      "description": "should accept legacy gssapiServiceName (GSSAPI)",
+      "uri": "mongodb://user%40DOMAIN.COM@localhost/?authMechanism=GSSAPI&gssapiServiceName=other",
+      "hosts": null,
+      "valid": true,
+      "warning": false,
+      "auth": {
+        "username": "user@DOMAIN.COM",
+        "password": null,
+        "db": "$external"
+      },
+      "options": {
+        "authmechanism": "GSSAPI",
+        "authmechanismproperties": {
+          "SERVICE_NAME": "other"
+        }
+      }
+    },
+    {
+      "description": "should throw an exception if provided with an invalid authSource (GSSAPI)",
+      "uri": "mongodb://user%40DOMAIN.COM@localhost?authMechanism=GSSAPI&authSource=foo",
+      "hosts": null,
+      "valid": false,
+      "warning": false,
+      "auth": null,
+      "options": null
+    },
+    {
+      "description": "should throw an exception when an authMechanism (GSSAPI) has an authSource other than $external",
+      "uri": "mongodb://user%40DOMAIN.COM@localhost/foo?authMechanism=GSSAPI&authSource=bar",
+      "hosts": null,
+      "valid": false,
+      "warning": true,
+      "auth": null,
+      "options": null
+    },
+    {
+      "description": "should throw an exception when an authMechanism (GSSAPI) is specified with no username",
```

Comment:
Get rid of "an authMechanism". MONGODB-X509 no longer requires a username, for example.

Reply:
Done